updater/internal/path/path_linux.go

```diff
@@ -15,24 +15,112 @@
 package path
 
 import (
+	"bufio"
+	"fmt"
+	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
 	"go.uber.org/zap"
 )
 
-// LinuxInstallDir is the install directory of the collector on linux.
-const LinuxInstallDir = "/opt/observiq-otel-collector"
+// DefaultLinuxInstallDir is the install directory of the collector on linux.
+const DefaultLinuxInstallDir = "/opt/observiq-otel-collector"
 
 // SystemdFilePath is the path for systemd service
 const SystemdFilePath = "/usr/lib/systemd/system/observiq-otel-collector.service"
 
 // SysVFilePath is the path for sysv service
 const SysVFilePath = "/etc/init.d/observiq-otel-collector"
 
+// DefaultConfigOverrides is a list of config files that can be used to override
+// package installation behavior. The updater needs to respect these config
+// options, such as BDOT_CONFIG_HOME
+var DefaultConfigOverrides = []string{
+	"/etc/default/observiq-otel-collector",
+	"/etc/sysconfig/observiq-otel-collector",
+}
+
+const (
+	configOverrideEnvVar = "BDOT_CONFIG_HOME"
+)
+
 // InstallDir returns the filepath to the install directory
-func InstallDir(_ *zap.Logger) (string, error) {
-	return LinuxInstallDir, nil
+func InstallDir(logger *zap.Logger, configOverrides []string) (string, error) {
```

Comment:
InstallDir was re-written to support reading the BDOT_CONFIG_HOME option. `path.InstallDir` is called from outside of this package, used to determine where the agent is installed.